backend/search_handler.py

```diff
@@ -21,21 +28,53 @@ def handler(event, context):
         }
     elif event['httpMethod'] == 'POST':
-        body = event.get('body')
-        print(body)
-        success = True
-        try:
-            response = run_query(es, 'dataall-index', body)
-        except Exception:
-            success = False
-            response = {}
-        return {
-            'statusCode': 200 if success else 400,
-            'headers': {
-                'content-type': 'application/json',
-                'Access-Control-Allow-Origin': '*',
-                'Access-Control-Allow-Headers': '*',
-                'Access-Control-Allow-Methods': '*',
-            },
-            'body': json.dumps(response),
-        }
+        if 'authorizer' in event['requestContext']:
+            if 'claims' not in event['requestContext']['authorizer']:
+                claims = event['requestContext']['authorizer']
+            else:
+                claims = event['requestContext']['authorizer']['claims']
+
+            # Needed for custom groups
+            user_id = claims['email']
+            if 'user_id' in event['requestContext']['authorizer']:
+                user_id = event['requestContext']['authorizer']['user_id']
+
+            groups = []
+            if os.environ.get('custom_auth', None):
+                groups.extend(get_custom_groups(user_id))
+            else:
+                groups.extend(get_cognito_groups(claims))
+
+            # Check if maintenance window is enabled AND if the maintenance mode is NO-ACCESS
+            if config.get_property('modules.maintenance.active'):
+                if (
+                    (MaintenanceService._get_maintenance_window_mode(engine=ENGINE) == MaintenanceModes.NOACCESS.value)
+                    and (
+                        MaintenanceService.get_maintenance_window_status(engine=ENGINE).status
+                        is not MaintenanceStatus.INACTIVE.value
+                    )
+                    and not TenantPolicyValidationService.is_tenant_admin(groups)
+                ):
+                    send_unauthorized_response(
+                        query={'operationName': 'OpensearchIndex'},
+                        message='Access Restricted: data.all is currently undergoing maintenance, and your actions are temporarily blocked.',
+                    )
+
+            body = event.get('body')
+            print(body)
+            success = True
+            try:
+                response = run_query(es, 'dataall-index', body)
+            except Exception:
+                success = False
+                response = {}
+            return {
+                'statusCode': 200 if success else 400,
+                'headers': {
+                    'content-type': 'application/json',
+                    'Access-Control-Allow-Origin': '*',
+                    'Access-Control-Allow-Headers': '*',
+                    'Access-Control-Allow-Methods': '*',
+                },
+                'body': json.dumps(response),
+            }
```

Comment:
It looks almost like the api handler, as next steps for the team: should we remove the search handler completely and integrate it into the api handler?

Reply:
Yes, the API handler and Search Handler are very similar. I agree we should integrate them into one as next steps. 
I can create a GH issue for this change. 

Not including this change in this PR as it might involve multiple infra changes in the code.

Reply:
Yes, but let's create an issue

Reply:
Created GH issue - https://github.com/data-dot-all/dataall/issues/1302